### DISCUSSION TITLE
Current State of SQLite Support

### DISCUSSION THREAD

**Discussion by PadraigK** [sqlc-dev/sqlc#1392] [Q&A]
Thanks for all your work on this project! It's exactly what I was looking for.

I've followed the comments in https://github.com/kyleconroy/sqlc/issues/161 and I've seen SQLite support referred to elsewhere as "experimental". So my understanding is that it kinda works, but isn't all there. 

Is sqlc usable with SQLite as is, with some minor missing details, or is it fully not a good idea to do that right now?

**Comment by mrjawright** [comment]
I just saw your message while looking to see if anyone else is using this for sqlite. ~~If you haven't tried it, yet, it doesn't seem to be working~~.  After getting everything set up, the generate gives me a "duplicate query name" error on the first query in my file every time...no matter what I change it to. I ran through the quick tutorial, and it looks promising once sqlite support gets out of beta.
Edit:
It seems that you have to group all of the queries for a given table together in the query file in order for it to parse correctly.
